src/AudioManager.ts

```diff
@@ -0,0 +1,158 @@
+import MediaAudio from "./MediaAudio";
+import AudioInputsService from "./AudioInputsService";
+import AudioOutputsService, {
+  AudioOutputsServicePlayOptions,
+} from "./AudioOutputsService";
+import AudioInputsBrowser from "./AudioInputsBrowser";
+
+export interface AudioManagerOptions {
+  duckVolumeLevel?: number;
+  normalVolumeLevel?: number;
+  sttService?: "browser" | "charisma/deepgram";
+  streamTimeslice?: number;
+  handleStartSTT: () => void;
+  handleStopSTT: () => void;
+  handleTranscript: (transcript: string) => void;
+  handleError?: (error: string) => void;
+}
+
+class AudioManager {
+  private audioInputsService: AudioInputsService;
+
+  private audioInputsBrowser: AudioInputsBrowser;
+
+  private audioOutputsService: AudioOutputsService;
+
+  private mediaAudio: MediaAudio;
+
+  private options: AudioManagerOptions;
+
+  constructor(options: AudioManagerOptions) {
+    this.options = {
+      duckVolumeLevel: options.duckVolumeLevel ?? 0,
+      normalVolumeLevel: options.normalVolumeLevel ?? 1,
+      sttService: options.sttService ?? "charisma/deepgram",
+      streamTimeslice: options.streamTimeslice ?? 100,
+      handleStartSTT: options.handleStartSTT,
+      handleStopSTT: options.handleStopSTT,
+      handleTranscript: options.handleTranscript,
+      handleError:
```

Comment:
It creates problems using a single type for both the constructor's options and options on the class, like on 56 the '?'. Suggest you assign the values you need on the class so we know they aren't undefined. Some you don't need, like where you use streamTimeslice on line 45, you don't need to keep it for anything onn AudioManager